tests/PySys/software-management-end-to-end/environment_sm_management.py

```diff
@@ -0,0 +1,180 @@
+import base64
+import time
+from datetime import datetime, timedelta, timezone
+
+import json
+import requests
+
+import pysys
+from pysys.basetest import BaseTest
+
+
+"""
+This environment provides an interface to software management features
+through the C8y REST API.
+With these we can emulate a user doing operations in the C8y UI.
+They are rather slow as they use the complete chain from end to end.
+
+WARNING: Handle with care!!!
+The C8YDEVICEID will handle on which this test will install and remove packages.
+"""
+
+
+def is_timezone_aware(stamp):
+    """determine if object is timezone aware or naive
+    See also: https://docs.python.org/3/library/datetime.html?highlight=tzinfo#determining-if-an-object-is-aware-or-naive
+    """
+    return stamp.tzinfo is not None and stamp.tzinfo.utcoffset(stamp) is not None
+
+
+class SmManagement(BaseTest):
+
+    PAGE_SIZE = "500"
+
+    def setup(self):
+        """Setup Environment"""
+
+        if self.myPlatform != "container":
+            self.skipTest("Testing the apt plugin is not supported on this platform")
+
+        tenant = self.project.tenant
+        user = self.project.username
+        password = self.project.c8ypass
+
+        auth = bytes(f"{tenant}/{user}:{password}", "utf-8")
+        self.header = {
+            b"Authorization": b"Basic " + base64.b64encode(auth),
+            b"content-type": b"application/json",
+            b"Accept": b"application/json",
+        }
+
+    def trigger_action(self, package_name, package_id, version, url, action):
+        """Trigger a installation or deinstallation of a package"""
+
+        self.trigger_action_json(
+            [
+                {
+                    "id": package_id,
+                    "name": package_name,
+                    "version": version,
+                    "url": url,
+                    "action": action,
+                }
+            ]
+        )
```

Comment:
It's a minor. In the argument, the order is `package_name`, then `package_id`, but in the JSON structure, the order is `package_id` then `package_name`. This actually confused me.